### ISSUE TITLE
npo.nl unknow url type npo+drm

### ISSUE THREAD

**Issue by arievanboxel** [rg3/youtube-dl#15709]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2018.02.26*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2018.02.26**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add the `-v` flag to **your command line** you run youtube-dl with (`youtube-dl -v <your command line>`), copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):

```
youtube-dl -v https://www.npo.nl/de-luizenmoeder/25-02-2018/AT_2091794
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'https://www.npo.nl/de-luizenmoeder/25-02-2018/AT_2091794']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2018.02.26
[debug] Python version 2.7.12 (CPython) - Linux-4.2.8-x86_64-with-glibc2.2.5
[debug] exe versions: ffmpeg 2.8.5
[debug] Proxy map: {}
[npo] AT_2091794: Downloading JSON metadata
[npo] AT_2091794: Downloading token
[npo] AT_2091794: Downloading formats JSON
[npo] AT_2091794: Downloading Adaptive stream JSON
[npo] AT_2091794: Downloading m3u8 information
[npo] AT_2091794: Downloading Hoog stream JSON
[npo] AT_2091794: Downloading Normaal stream JSON
[npo] AT_2091794: Downloading Laag stream JSON
[debug] Default format spec: bestvideo+bestaudio/best
[debug] Invoking downloader on u'npo+drm://internetvod.omroep.nl/AT_2091794'
ERROR: unable to download video data: <urlopen error unknown url type: npo+drm>
Traceback (most recent call last):
  File "/opt/lib/python2.7/site-packages/youtube_dl/YoutubeDL.py", line 1896, in process_info
    success = dl(filename, info_dict)
  File "/opt/lib/python2.7/site-packages/youtube_dl/YoutubeDL.py", line 1835, in dl
    return fd.download(name, info)
  File "/opt/lib/python2.7/site-packages/youtube_dl/downloader/common.py", line 364, in download
    return self.real_download(filename, info_dict)
  File "/opt/lib/python2.7/site-packages/youtube_dl/downloader/http.py", line 340, in real_download
    establish_connection()
  File "/opt/lib/python2.7/site-packages/youtube_dl/downloader/http.py", line 109, in establish_connection
    ctx.data = self.ydl.urlopen(request)
  File "/opt/lib/python2.7/site-packages/youtube_dl/YoutubeDL.py", line 2199, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/opt/lib/python2.7/urllib2.py", line 429, in open
    response = self._open(req, data)
  File "/opt/lib/python2.7/urllib2.py", line 452, in _open
    'unknown_open', req)
  File "/opt/lib/python2.7/urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "/opt/lib/python2.7/urllib2.py", line 1266, in unknown_open
    raise URLError('unknown url type: %s' % type)
URLError: <urlopen error unknown url type: npo+drm>
```

---

### Description of your *issue*, suggested solution and other information

NPO has probably changed their format. Since 3 days programs via this site do not download anymore.

**Comment by jpluimers** [comment]
I understand that you are busy, but I'm busy too and still invested a few hours finding out where problems started and what could be related. 

Why? Because it is open source and open source projects can use any help they can get.

So where is the original issue that is not a duplicate and not closed? For many, that's the only way to properly track progress and certainly not warrants a rude

> Stop posting duplicates and wait for the next release.

It certainly is not just a breakage on the npo side, as https://github.com/rg3/youtube-dl/releases/tag/2018.02.11 works but https://github.com/rg3/youtube-dl/releases/tag/2018.02.22 fails.

I bit the bullet and did a `git bisect` based on https://stackoverflow.com/questions/4713088/how-to-use-git-bisect/4714297#4714297

So at least in part this is a regression, as this is the commit that started failing: https://github.com/rg3/youtube-dl/commit/ee706f1009a90e9911f31d01aa9eae41cc34e75b

```
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[ee706f1009a90e9911f31d01aa9eae41cc34e75b] [npo] Improve quality metadata extraction
```

**Comment by dstftw** [comment]
Once again: it's a breakage at **npo side** due to **introducing `npo+drm` URL scheme** after 2018.02.26 was released that is already fixed in d1b61870127fdee53147ac4c249af34a35562dbe. Regression is when something breakes due to **your** changes not due to changes of 3rdparties.
